mmgen/core/evaluation/metrics.py

```diff
@@ -16,12 +17,116 @@
 from mmgen.models.architectures import InceptionV3
 from mmgen.models.architectures.common import get_module_device
 from mmgen.models.architectures.lpips import PerceptualLoss
+from mmgen.utils.io_utils import download_from_url
 from ..registry import METRICS
 from .metric_utils import (_f_special_gauss, _hox_downsample,
                            compute_pr_distances, finalize_descriptors,
                            get_descriptors_for_minibatch, get_gaussian_kernel,
                            laplacian_pyramid, slerp)
 
+INCEPTION_URL = 'https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/inception-2015-12-05.pt'  # noqa
+
+
+def load_inception(inception_args, metric):
+    """Load Inception Model from given ``inception_args`` and ``metric``. This
+    function would try to load Inception under the guidance of 'type' given in
+    `inception_args`, if not given, we would try best to load Tero's ones. In
+    detailly, we would first try to load model from disk with given
+    'inception_path', and then try to download the checkpoints from
+    'inception_url'. If both method are failed, pytorch version of Inception
+    would be loaded.
+
+    Args:
+        inception_args (dict): Keyword args for inception net.
+        metric (string): Metric to use the Inception. This argument would
+            influence the pytorch's Inception loading.
+    Returns:
+        model (torch.nn.Module): Loaded Inception model.
+        style (string): The version of the loaded Inception.
+    """
+
+    # load pytorch version if inception_args is invalid
+    if not isinstance(inception_args, dict):
+        mmcv.print_log(
+            'Receive invalid \'inception_args\': '
+            f'{inception_args}, load pytorch version.', 'mmgen')
+        return _load_inception_torch(inception_args, metric), 'pytorch'
+
+    _inception_args = deepcopy(inception_args)
+    if 'type' in inception_args:
+        inceptoin_type = _inception_args.pop('type')
+    else:
+        inceptoin_type = ''
+
+    # load pytorch version
+    if inceptoin_type == 'pytorch':
+        return _load_inception_torch(_inception_args, metric), 'pytorch'
+
+    # try to load Tero's version
+    path = _inception_args.get('inception_path', '')
+    url = _inception_args.get('inception_url', INCEPTION_URL)
+
+    model = _load_inception_from_path(path)
+    if isinstance(model, torch.nn.Module):
```

Comment:
This design is very confusing. What's the behavior of  `_load_inception_from_path('')`?

Reply:
`_load_inception_from_path` would fail, and then `_load_inception_from_url` would be called to download the checkpoints.